### PR TITLE
Mega Carp Re-Carpening: Carpatastrophe! 

### PR DESCRIPTION
**Pull request by Amelia0010**
port of https://github.com/tgstation/tgstation/pull/31365

#### FIRST CHANGE: RANDOM NAMES FOR MEGACARP!

That's The Unique Joe.
That's Man Bear Goblin Shark.
That's Scarcely Lethal Toothy.
That's Sarcastic Chompy.
(I'll probably just make these toothy chompy and all that other kind of stuff)

#### SECOND CHANGE: Healing!

slow healing when out of combat

#### THIRD CHANGE: Random Stat Sharks!

they get random stats when they are created, it'll be fun trying to make a legendary shark as a xenobiologist.

#### Changelog

:cl:  
rscadd: Megacarps will now have random names and stats, and can now heal when not fighting.
/:cl:

### CODE REVIEW

**Comment by Cruix** [comment]
This is just a blatant attempt to add a special snowflake overpowered simple mob for xenobiology to make sentient.

**Comment by ShadowDeath6** [comment]
make it lose the buffs when player controlled so xenobio doesn't have even more shit to mess with

File: code/modules/mob/living/simple_animal/hostile/carp.dm [hunk 1]
```diff
@@ -58,19 +60,40 @@
...
 	melee_damage_lower = 20
 	melee_damage_upper = 20
 
+	var/regen_cooldown = 0
+
+/mob/living/simple_animal/hostile/carp/megacarp/New()
+	. = ..()
+	name = "[pick(megacarp_first_names)] [pick(megacarp_last_names)]"
+	melee_damage_lower += rand(2, 10)
+	melee_damage_upper += rand(10, 20)
```
Comment (ShadowDeath6) [inline code/modules/mob/living/simple_animal/hostile/carp.dm:77]: so this can do upper damage of 40 provided they get good rolls?

**Comment by Amelia0010** [comment]
I'll make them lose their buffs if player controlled tomorrow when I get back to a computer that doesn't need 30 minutes to compile and launch a server

**Comment by drderp3635** [comment]
no one would ever search for stats for a non player controlled carp though

**Comment by Amelia0010** [comment]
no but it actually makes megacarp special rather than just "a slightly larger carp"

EDIT: what about not making them lose all of the buff when they're sentient, but just less powerful?

**Comment by X-TheDark** [comment]
But why? I'm pretty sure this mob is useless for anyone other than traitor Xenobiologists and they usually have better things to do than spend the entire round trying to make this mob.

I like the idea of variation, but just limiting it to "more health" and "more damage" seems like an odd choice.

I would understand if you'd add some carp that can be useful for everyone, not just traitors (akin to Revival Magicarp - R.I.P.).

**Comment by Amelia0010** [comment]
This PR is bad and I should feel bad

**Comment by Time-Green** [comment]
dammit I liked this

**Comment by Time-Green** [comment]
nich please

**Comment by Time-Green** [comment]
we can overthrow them

**Comment by Amelia0010** [comment]
meh, I need to come up with something better for spess

**Comment by 1fbff5f83b23d39d38b1dfcb4cac8d9b** [comment]
@ Tribunal overturn this

**Pull request closed without merging**